bbc/simorgh
# WSTEAM1-679 - Surface 404 for Summary Key Points Article Pages

Resolves JIRA [WSTEAM1-702](https://jira.dev.bbc.co.uk/browse/WSTEAM1-702)

Overall changes
======
Adds extra check within `shouldRender()` that checks if a specific `tagging` exists within an Article page. This tagging is set manually by editorial to indicate that the page is a Key/Summary Points page. If this tag exists, then we should return a 404 for the page instead of rendering it.

The logic within shouldRender has been refactored a little to make it a little bit more understandable/readable.

Code changes
======

- Adds `tagging` check logic to shouldRender()
- Re-arranges logic in shouldRender to make it more readable
- Adds and updates unit tests for new `tagging` case

Testing
======
1. Visit a Summary key points asset locally: http://localhost:7080/pidgin/articles/c65x3zwr3gro?renderer_env=test
2. You should be served a 404 when viewing it
3. Visit a normal article page locally: http://localhost:7080/kyrgyz/articles/c51ey5rkdkxo?renderer_env=live
4. You should see the article as normal

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)
